frontend/src/utils.ts

```diff
@@ -116,3 +116,85 @@ export class Emitters {
   // Coffee Chats
   static coffeeChatsUpdated: EventEmitter<void> = new EventEmitter();
 }
+
+/**
+ * Parses links from english text and fixes a provided link to ensure it adheres to a proper format for LinkedIn or GitHub profiles.
+ * Also handles raw usernames (e.g., "jujulcrane") by constructing a proper URL.
+ *
+ * @param {string} link - The input link, which could be a full URL, partial URL, or raw username.
+ * @param {boolean} git - Whether the link is for GitHub. If true, it processes the link as a GitHub URL.
+ * @param {boolean} linkedIn - Whether the link is for LinkedIn. If true, it processes the link as a LinkedIn URL.
+ * @returns {string | undefined} - The fixed and properly formatted link or undefined if the input is invalid.
+ *
+ * @example
+ * fixLink("https://linkedin.com/in/jujulcrane", false, true);
+ * // Returns: "https://www.linkedin.com/in/jujulcrane"
+ *
+ * @example
+ * fixLink("github.com/jujulcrane", true, false);
+ * // Returns: "https://github.com/jujulcrane"
+ *
+ * @example
+ * fixLink("Visit my profiles: https://linkedin.com/in/jujulcrane and github.com/jujulcrane", false, true);
+ * // Returns: undefined (multiple links in the input)
+ *
+ * @example
+ * fixLink("jujulcrane", false, true);
+ * // Returns: "https://www.linkedin.com/in/jujulcrane/"
+ */
+export const fixLink = (link: string, git: boolean, linkedIn: boolean): string | undefined => {
```

Comment:
Maybe instead of booleans, we could change this into just a parameter ("linkType")? That way if we want to support more links, we won't have to add an extra boolean for each one (also I think it'll make it easier to read). You can just make a `type LinkType = "git" | "linkedIn"` and use that type here.

Also super small/opinionated nit but I think `formatLink` might be a better name here. "fix" feels like it's fixing a bug.